app/Application.php

```diff
@@ -138,28 +144,40 @@ protected function finishApp()
 
     protected function showHistory()
     {
-        $history = $this->historyRepository->all();
-
-        if (empty($history)) {
+        if (!($this->historyRepository->isExist())) {
             return info($this->messages['info']['no_history']);
         }
 
         do {
-            $historyGroups = array_group($history, 'date');
-            $historyCommands = array_merge([FULL, 'help', 'back'], array_keys($historyGroups));
-            $showDateHistory = ask($this->getText('info', 'info_history', FULL));
-            $isDataValid = (is_date($showDateHistory) || in_array($showDateHistory, $historyCommands));
+            $output = choice('You want show history on screen or export in file: ', [EXPORT_HISTORY, SCREEN]);
 
-            if (!$isDataValid) {
-                info($this->messages['errors']['history_wrong_input']);
+            if ($output === 'export') {
+                $nameOfFile = readline("{$this->messages['info']['name_of_file_create']}");
+                $pathToFile = readline("{$this->messages['info']['name_of_directory_create']}");
 
-                continue;
+                $fullPathName = "{$pathToFile}{$nameOfFile}.txt";
+
+                $this->historyTxtExporter = new HistoryTxtExporter($fullPathName);
```

Comment:
you've already created new instance of the history repository class inside the constructor. To set filepath you can implement the setter method

```suggestion
                $this->historyTxtExporter->setFilePath($fullPathName);
```